spin/sessions.go

```diff
@@ -0,0 +1,35 @@
+package spin
+
+import (
+	uh "github.com/TRON-US/go-btfs/core/commands/storage/upload/helper"
+	"github.com/TRON-US/go-btfs/core/commands/storage/upload/sessions"
+	"github.com/TRON-US/go-btfs/core/commands/storage/upload/upload"
+
+	cmds "github.com/TRON-US/go-btfs-cmds"
+
+	"go4.org/syncutil"
+)
+
+func RenterSessions(req *cmds.Request, env cmds.Environment) {
+	params, err := uh.ExtractContextParams(req, env)
+	if err != nil {
+		return
+	}
+
+	cursor, err := sessions.GetRenterSessionsCursor(params)
+	if err != nil {
+		return
+	}
+
+	sem := syncutil.NewSem(10)
+	session, err := cursor.NextSession(sessions.RssWaitUploadReqSignedStatus)
+	for err == nil && session != nil {
+		go func(session *sessions.RenterSession) {
+			sem.Acquire(1)
+			defer sem.Release(1)
+			if e := upload.ResumeWaitUploadOnSigning(session); e != nil {
+				return
+			}
+		}(session)
```

Comment:
There needs to be some sem wait here?

Reply:
the call will block if the available units falls below zero. so maybe no need to wait.